src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int figureCount = 5;
+    private final int randomFigureParameter = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomFigureNum = random.nextInt(figureCount);
+        switch (randomFigureNum) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(),
```

Comment:
Don't need to call colorSupplier.getRandomColor() each case, create a local variable